packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +290,102 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed']
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def query_dns_server(fqdn, zone, nameserver_ip)
+      query_dns = Resolv::DNS.new(nameserver: [nameserver_ip], search: [zone], ndots: 1)
+      a_record = query_dns.getresource(fqdn, Resolv::DNS::Resource::IN::A)
```

Comment:
In Foreman 2.1 we'll also set set AAAA records (https://github.com/theforeman/puppet-foreman_proxy/commit/fecf9c8b8bc3a50bdc0ace5e96840fae90c9d240)

Reply:
@ekohl I created a separate [issue](https://projects.theforeman.org/issues/29268) so this can be addressed in a follow-up.

Reply:
The new update_zone addresses this.